src/main/java/Storage.java

```diff
@@ -0,0 +1,108 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+
+/**
+ * Storage class handles loading and saving of tasks into an external file.
+ */
+public class Storage {
+    String filePath;
+    File file;
+
+    public Storage(String filePath) {
+        file = new File(filePath);
+        try {
+            file.getParentFile().mkdir();
+            file.createNewFile();
+        } catch (IOException e) {
+            System.err.println("Unable to create file");
+        }
+    }
+
+    /**
+     * Loads tasks data from file.
+     *
+     * @return Array list of Task.
+     * @throws FileNotFoundException If file fails to load.
+     */
+    public ArrayList<Task> loadData() throws FileNotFoundException {
+        ArrayList<Task> dukeList = new ArrayList<>();
+        Scanner sc = new Scanner(file);
+        Task t;
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            String[] splitBySpace = line.split(" ");
+            String[] splitBySlash;
+            String command = splitBySpace[0];
+
+            if (command.equals("D")) {
+                splitBySlash = line.split("/");
+
+                //Splits second time
```

Comment:
Comment can be more descriptive. E.g. explain why do we need to split second time